R/shared-methods.R

```diff
@@ -134,13 +137,13 @@ DataClass <- R6::R6Class(
     supported_region_names = list("1" = NA),
     #' @field supported_region_codes A list of region codes in order of level.
     supported_region_codes = list("1" = NA),
-    #' @field region_name string Name for the codes column, e.g. 'iso_3166_2'
+    #' @field region_name string Name for the region column, e.g. 'region'
     region_name = NULL,
     #' @field code_name string Name for the codes column, e.g. 'iso_3166_2'
     code_name = NULL,
     #' @field codes_lookup string or tibble Region codes for the target country
     codes_lookup = list(),
-    #' @field data_urls List of named links to raw data.
+    #' @field data_urls List of named url links to raw data.
     data_urls = list(),
```

Comment:
Should this field (and some others below) be marked as private, or should it at least be documented as something which the initialisation code is going to overwrite?

Reply:
Private attributes and methods don't get documented with Roxygen. I lean more towards keeping everything public so everything is traceable in the docs.

I'm open to dicussing ideas about this though (we could for example have it all explained separately and just leave the docs as need to know user functions)

Reply:
Sorry, you have explained this to me once already. Let's please emphasize that this (and the others below) are overwritten by other bits of code so class authors shouldn't try writing to them or initialising them themselves.